### PR TITLE
fixes #264. fixes #277

### LINKED ISSUE #264: Comment + else/elseif blocks
as3 code for example:
```actionscript
package {
  public class Bug {
    public function Bug() {
      
      var obj:Object = new Object();
      var message:String;
      
      // error inside
      if (obj.error is Error) {
        message = obj.error.message;
      }
      // error event inside
      else if (obj.error is ErrorEvent) {
        message = obj.error.text;
      }
      // unknown
      else {
        message = obj.error.toString();
      }
      
    }
  }
}
```
expected result
```haxe
class Bug {
  public function new() {
    var obj:Dynamic = {};
    var message:String;
    
    // error inside
    if (Std.is(obj.error, Error)) {
      message = obj.error.message;
    }
    // error event inside
    else if (Std.is(obj.error, ErrorEvent)) {
      message = obj.error.text;
    }
    // unknown
    else {
      message = Std.string(obj.error);
    }
  }
}
```
actual result
```haxe
class Bug {
  public function new() {
    var obj:Dynamic = {};
    var message:String;
    
    // error inside
    if (Std.is(obj.error, Error)) {
      message = obj.error.message;
    }
    else {
      // error event insideif (Std.is(obj.error, ErrorEvent)) {
        message = obj.error.text;
      }
      else {
        // unknown{
          message = Std.string(obj.error);
        }
      }
    }
  }
}
```

### LINKED ISSUE #277: else if not properly nested
as3 code for example:
```actionscript
package {
    public class Issue {
        public function Issue() {
            if(1) {
            	a();
            }else if(2) {
            	b();
            }else if(3) {
                c();
            }else if(4) {
                d();
            }else if(5) {
                e();
            }else if(6) {
                f();
            }else if(7) {
                g();
            }else if(8) {
                h();
            }else if(9) {
                i();
            }else if(10) {
                j();
            }else{
                k();
            }
        }
    }
}
```
expected result
```haxe
package {
    public class Issue {
        public function Issue() {
            if(1) {
                a();
            }else if(2) {
                b();
            }else if(3) {
                c();
            }else if(4) {
                d();
            }else if(5) {
                e();
            }else if(6) {
                f();
            }else if(7) {
                g();
            }else if(8) {
                h();
            }else if(9) {
                i();
            }else if(10) {
                j();
            }else{
                k();
            }
        }
    }
}
```
actual result
```haxe
class Issue
{
    public function new()
    {
        if (1)
        {
            a();
        }
        else
        {
            if (2)
            {
                b();
            }
            else
            {
                if (3)
                {
                    c();
                }
                else
                {
                    if (4)
                    {
                        d();
                    }
                    else
                    {
                        if (5)
                        {
                            e();
                        }
                        else
                        {
                            if (6)
                            {
                                f();
                            }
                            else
                            {
                                if (7)
                                {
                                    g();
                                }
                                else
                                {
                                    if (8)
                                    {
                                        h();
                                    }
                                    else
                                    {
                                        if (9)
                                        {
                                            i();
                                        }
                                        else
                                        {
                                            if (10)
                                            {
                                                j();
                                            }
                                            else
                                            {
                                                k();
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```

### PR DESCRIPTION
**Pull request by mastef**
- keeps `else if` statements in one line
- handles comments between conditional blocks
- removes double-nested blocks

### CODE REVIEW

File: src/as3hx/Writer.hx [hunk 1]
```diff
@@ -1821,9 +1827,45 @@ class Writer
...
+                        }
+                    }
+                case EIf(_, _, _):
+                    elseif = e2;
+                default:
+            }
+            if(elseif != null) {
+                writeIndent("else ");
+                result = writeExpr(elseif);
+            }else{
```
Comment (SlavaRa) [inline src/as3hx/Writer.hx]: please replace `}else{` to `} else { `

File: src/as3hx/Writer.hx [hunk 2]
```diff
@@ -126,6 +126,12 @@ class Writer
                         if (i == 0) f(ex[0]);
                         else result.push(ex[i]);
                     }
+                case ECommented(s,b,t,e):
+                    // issue 264
```
Comment (SlavaRa) [inline src/as3hx/Writer.hx]: please remove this comment

File: src/as3hx/Writer.hx [hunk 3]
```diff
@@ -1821,9 +1827,45 @@ class Writer
             e2 = f(e2);
             e2 = EBlock(formatBlockBody(e2));
             writeNL();
-            writeIndent("else");
-            writeStartStatement();
-            result = writeExpr(e2);
+            var elseif:Expr = null;
+            // issue 277
```
Comment (SlavaRa) [inline src/as3hx/Writer.hx]: please remove this comment

File: src/as3hx/Writer.hx [hunk 4]
```diff
@@ -1821,9 +1827,45 @@ class Writer
...
+                            case ENL(e4):
+                                switch(e4) {
+                                    case EIf(_, _, _):
+                                        // found single if statement after an else
+                                        // replace parent `block` + `new line` with
+                                        // the `if` statement instead so we stay on
+                                        // the same line as the `else` -> `else if`
+                                        elseif = e4;
+                                    case EBlock(_):
+                                        // issue 264
```
Comment (SlavaRa) [inline src/as3hx/Writer.hx]: please remove this comment

**Comment by SlavaRa** [comment]
And please update CHANGES.md

**Comment by mastef** [comment]
Ok! @SlavaRa all done

**Comment by mastef** [comment]
🎉  Thank you @SlavaRa ! I'm nearly through 273. What do you think about 282?

**Comment by SlavaRa** [comment]
I'll look #282 a little later and after that write my opinion

**Pull request merged**